src/core/tritonserver.h

```diff
@@ -834,6 +834,17 @@ TRITONSERVER_EXPORT TRITONSERVER_Error* TRITONSERVER_InferenceResponseId(
     TRITONSERVER_InferenceResponse* inference_response,
     const char** request_id);
 
+/// Check whether or not the response is from a model with decoupled
+/// requests and responses.
+///
+/// \param inference_response The response object.
+/// \param is_decoupled Returns the truth value of whether the response
+/// is decoupled or not.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_EXPORT TRITONSERVER_Error*
+TRITONSERVER_InferenceResponseIsDecoupled(
```

Comment:
I think we will likely need some function like this to help us understand some properties of the response, but to future-proof I think we will want to make it as general as possible. Can you hold off on this part of the change and include it with your grpc_server changes... we might have more insight into exactly what we need at that point.